libvast/vast/concept/parseable/core/choice.hpp

```diff
@@ -32,34 +32,32 @@ constexpr bool is_choice_parser_v = is_choice_parser<T>::value;
 /// Attempts to parse either LHS or RHS.
 template <class Lhs, class Rhs>
 class choice_parser : public parser<choice_parser<Lhs, Rhs>> {
-public:
+private:
   using lhs_attribute = typename Lhs::attribute;
   using rhs_attribute = typename Rhs::attribute;
 
   // LHS = unused && RHS = unused  =>  unused
-  // LHS = T && RHS = unused       =>  LHS
   // LHS = unused && RHS = T       =>  RHS
+  // LHS = T && RHS = unused       =>  LHS
   // LHS = T && RHS = T            =>  T
   // LHS = T && RHS = U            =>  variant<T, U>
-  using attribute =
-    std::conditional_t<
-      std::is_same_v<lhs_attribute, unused_type>
-        && std::is_same_v<rhs_attribute, unused_type>,
-      unused_type,
-      std::conditional_t<
-        std::is_same_v<lhs_attribute, unused_type>,
-        rhs_attribute,
-        std::conditional_t<
-          std::is_same_v<rhs_attribute, unused_type>,
-          lhs_attribute,
-          std::conditional_t<
-            std::is_same_v<lhs_attribute, rhs_attribute>,
-            lhs_attribute,
-            detail::flattened_variant<lhs_attribute, rhs_attribute>
-          >
-        >
-      >
-    >;
+  static constexpr auto get_attribute() {
```

Comment:
We try to avoid `get_` and `set_` prefixes of function names where possible, because they don't add any insight. How about we call this one `attribute_type()`?

Reply:
Sure, just renamed it to `attribute_type`.

Reply:
Thanks!